tests/5.0/tasks/test_task_depend_iterator.cpp

```diff
@@ -0,0 +1,56 @@
+//===-- test_task_depend_iterator.cpp - test of the iterator clause inside the task depend clause ----===//
+// 
+// OpenMP API Version 5.0 Nov 2020
+// 
+// This file is a test for the iterator modifier when used with the task depend
+// clause. This modifier should create an iterator that expand into multiple values
+// inside the clause they appear. In this particular test case the iterator expands into
+// several values creating several dependencies at the same time.
+//
+//===-------------------------------------------------------------------------------------------------===//
+
+#include <omp.h>
+#include <algorithm>
+#include <chrono>
+#include <cstdlib>
+#include <iostream>
+#include <thread>
+#include <vector>
+#include "ompvv.h"
+
+int test_task_depend_iterator() {
+  int ptr[] = {0, 4, 5, 6, 7, 8, 9, 10, 11}, cols[] = {1, 2, 3, 4, 5, 5, 6, 6, 7, 7, 8};
+  std::vector<int> threadOrder;
+  bool threadOrderError = false;
+#pragma omp parallel num_threads(4)
```

Comment:
@jhdavis8 shall we also have a `OMPVV_HOST_NUM_THREADS`
?

Reply:
I don't see a reason why not. Although it is OK for tests to hard-code values as long as they are not unreasonable.

Reply:
The main reason of why it's hard coded to 4, is because the dependencies follows the graph:
```dot
digraph G {
   0 -> 1
   0 -> 2
   0 -> 3
   0 -> 4
   1 -> 5
   2 -> 5
   3 -> 6
   4 -> 6
   5 -> 7
   6 -> 7
}
````
Hence it could execute the spanning task from 0 all at the same time.

Reply:
Having 4 threads does not guarantee this, and having more threads does not imply otherwise.
4 is fine, and not unreasonable, but we have received complains about hardcoding the number of threads and teams. This is more on the context of devices. 

For this particular case, it does not really make much of a difference, I was just wondering if it makes sense to think of a host OMPVV_NUM_THREADS for host for the sake of our infrastructure.

Reply:
@josemonsalve2 There is, as of recently, a OMPVV_NUM_THREADS_HOST macro available for these use. It was added with the device num macros.

Reply:
Added the macro.